core/src/main/java/com/alibaba/fescar/core/rpc/netty/AbstractRpcRemotingServer.java

```diff
@@ -150,16 +150,30 @@ public void initChannel(SocketChannel ch) {
             ChannelFuture future = this.serverBootstrap.bind(listenPort).sync();
             LOGGER.info("Server started ... ");
             RegistryFactory.getInstance().register(new InetSocketAddress(XID.getIpAddress(), XID.getPort()));
+            registerShutdownHook();
             future.channel().closeFuture().sync();
         } catch (Exception exx) {
             throw new RuntimeException(exx);
         }
 
     }
 
+    /**
+     * regiter vm shutdown hook for graceful shutdown
+     * can't prevent kill -9
+     */
+    private void registerShutdownHook() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> shutdown()));
+    }
+
     @Override
     public void shutdown() {
         try {
+            LOGGER.info("Server shutdown ... ");
```

Comment:
```java
if (isEnableDebug()) {
    LOGGER.debug("Start to shutdown server. ");
}
```

Reply:
Why did the previous code use LOGGER.info("Server started ... ");

Reply:
I don't know. just think it's an unnecessary message.